src/rebar_prv_install_deps.erl

```diff
@@ -510,7 +512,7 @@ parse_goal(Name, Constraint) ->
         {match, [Op, Vsn]} ->
             {Name, Vsn, binary_to_atom(Op, utf8)};
         nomatch ->
-            fail
+            throw(?PRV_ERROR({bad_constraint, Name, Constraint}))
```

Comment:
Oh god, I think I got rid of all the `throw/1` calls before in my last patch. I'm amazed this shit still passed. Should we revert my own, @tsloughter ?